test/components/date-picker.spec.tsx

```diff
@@ -28,386 +34,313 @@ describe('The DatePicker Component', () => {
+        const {driver: datePicker} = clientRenderer.render(<DatePicker onChange={onChange} />)
+            .withDriver(DatePickerTestDriver);
 
-        trigger.change(datePickerInput!, '2sgsdfsdfw223');
-        simulate.blur(datePickerInput);
-
-        await waitForDom(() => expect(onChange).to.have.not.been.calledOnce);
+        datePicker.changeDate('2sgsdfsdfw223');
+        sleep(20);
+        expect(onChange).to.have.not.been.called;
     });
 
     it('should call onChange with the current input value when blurred', async () => {
         const onChange = sinon.spy();
-        const {select, waitForDom} = clientRenderer.render(<DatePicker value={JANUARY_FIRST} onChange={onChange} />);
-        const datePickerInput = select(datePickerInputId);
+        const {driver: datePicker, waitForDom} = clientRenderer.render(
+            <DatePicker value={JANUARY_FIRST} onChange={onChange} />
+        ).withDriver(DatePickerTestDriver);
 
-        trigger.change(datePickerInput!, '2017/02/01');
-        simulate.blur(datePickerInput);
+        datePicker.changeDate('2017/02/01');
 
         await waitForDom(() => expect(onChange).to.have.been.calledWithMatch({value: FEBRUARY_FIRST}));
     });
 
     it('should use a provided placeholder', async () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker placeholder="mm/dd/yyyy" />);
+        const {driver: datePicker, waitForDom} = clientRenderer.render(<DatePicker placeholder="mm/dd/yyyy" />)
+            .withDriver(DatePickerTestDriver);
 
-        await waitForDom(() => expect(select(datePickerInputId)).to.have.attribute('placeholder', 'mm/dd/yyyy'));
+        await waitForDom(() => expect(datePicker.input).to.have.attribute('placeholder', 'mm/dd/yyyy'));
     });
 
     it('should show and hide the dropdown when the input is clicked', async () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker />);
-        const datePickerInput = select(datePickerInputId);
-
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.absent());
-
-        simulate.mouseDown(datePickerInput);
-
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.present());
-
-        simulate.mouseDown(datePickerInput);
-
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.absent());
-    });
-
-    it('should show and hide the dropdown when the calendar icon is clicked', async () => {
-        const {select, waitForDom} = clientRenderer.render(<DatePicker />);
-        const calendarIcon = select('CALENDAR_ICON');
+        const {driver: datePicker, waitForDom} = clientRenderer.render(<DatePicker />)
+            .withDriver(DatePickerTestDriver);
 
-        await waitForDom(() => expect(bodySelect(datePickerDropdownId)).to.be.absent());
+        await waitForDom(() => expect(datePicker.isOpen()).to.be.false);
 
-        simulate.click(calendarIcon);
+        datePicker.click(datePicker.input!);
```

Comment:
Maybe have the input as default so you can write `datePicker.click()` instead of specifying the input?